test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -189,6 +193,19 @@ public static void assertBlocked(final ActionRequestBuilder builder, @Nullable f
         assertBlocked(builder, expectedBlock != null ? expectedBlock.id() : null);
     }
 
+    private static boolean checkRetryableBlock(Set<ClusterBlock> clusterBlocks){
+        boolean found = false;
```

Comment:
Since there will always be at least one block, we can simplify this to just bail out of the loop with false when an offending block is found and then just return true at the end. `found` can thus be removed.

You can add an assertion to make this assumption explicit in the code (is given by the buildMessageForXXXBlocks methods).

Reply:
OK, I will do that.

Reply:
In ElasticsearchAssertions.assertBlocked method, a assertion is given:
`assertThat(clusterBlockException.blocks().size(), greaterThan(0));`. So I think we do not need to add another assumption. I have removed the variable `found` yet.